includes/utilities.php

```diff
@@ -159,6 +166,9 @@ function insert_redirect( $from, $to, $status_code, $post_id = 0 ) {
 			'post_status'           => 'publish',
 			'post_title'            => strtolower( $from ),
 			'post_type'             => REDIRECTS_POST_TYPE,
+			'meta_input' => [
+				'preserve_parameters' => is_bool( $preserve_parameters ) ? $preserve_parameters : '',
```

Comment:
I think for post meta `false` equates to an empty string. What's the reason for having `null` as an option rather than defaulting to `false` whenever `$preserve_parameters` isn't set or is `false`?

You could check if it's truthy and then save a "0" or a "1". A better way to ensure that a variable is of a particular tyoe is to use type hinting in the function declaration e.g. `function ( bool $var = false )`

Reply:
> I think for post meta false equates to an empty string. What's the reason for having null as an option rather than defaulting to false whenever $preserve_parameters isn't set or is false?

NULL as an option will be surplus to requirement. I have removed it from the filter so now the `preserve_parameters` will either be `true` or `false`.

I have added the type hinting in the function declaration too.